### PR TITLE
Fix `CONFIG_ARCHIVE_MAP_MAPPING_NAMES`

### LINKED ISSUE #21204: Typo in XCLIP model
### System Info

transformers version 4.25.1

### Who can help?

@NielsRogge @ydshieh 

There's a typo/mismatch for the xclip model pretrained config archive map between https://github.com/huggingface/transformers/blob/main/src/transformers/models/auto/configuration_auto.py#L333

and. https://github.com/huggingface/transformers/blob/main/src/transformers/models/x_clip/configuration_x_clip.py#L27

notice in the first example, there is underscore between X and CLIP whereas in the second example, there isn't.

This leads to an error when initializing automodel

`   
    huggingface_models = list(ALL_PRETRAINED_CONFIG_ARCHIVE_MAP.keys())

  File "/usr/local/lib/python3.8/dist-packages/transformers/models/auto/configuration_auto.py", line 612, in keys

    self._initialize()

  File "/usr/local/lib/python3.8/dist-packages/transformers/models/auto/configuration_auto.py", line 602, in _initialize

    mapping = getattr(module, map_name)

  File "/usr/local/lib/python3.8/dist-packages/transformers/utils/import_utils.py", line 1086, in __getattr__

    raise AttributeError(f"module {self.__name__} has no attribute {name}")

AttributeError: module transformers.models.x_clip has no attribute X_CLIP_PRETRAINED_CONFIG_ARCHIVE_MAP`

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

1. from transformers import ALL_PRETRAINED_CONFIG_ARCHIVE_MAP
2. huggingface_models = list(ALL_PRETRAINED_CONFIG_ARCHIVE_MAP.keys())

### Expected behavior

Expect no errors to be raised

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Fix `CONFIG_ARCHIVE_MAP_MAPPING_NAMES` as reported in #21204.
Also, `UPERNET_PRETRAINED_CONFIG_ARCHIVE_MAP` doesn't exist.

Remark: we have planned deprecation
```
        warnings.warn(
            "ALL_PRETRAINED_CONFIG_ARCHIVE_MAP is deprecated and will be removed in v5 of Transformers. "
            "It does not contain all available model checkpoints, far from it. Checkout hf.co/models for that.",
            FutureWarning,
        )
```

Fix #21204

### CODE REVIEW

File: src/transformers/models/auto/configuration_auto.py [hunk 1]
```diff
@@ -318,7 +318,6 @@
         ("transfo-xl", "TRANSFO_XL_PRETRAINED_CONFIG_ARCHIVE_MAP"),
         ("unispeech", "UNISPEECH_PRETRAINED_CONFIG_ARCHIVE_MAP"),
         ("unispeech-sat", "UNISPEECH_SAT_PRETRAINED_CONFIG_ARCHIVE_MAP"),
-        ("upernet", "UPERNET_PRETRAINED_CONFIG_ARCHIVE_MAP"),
```
Comment (ydshieh) [inline src/transformers/models/auto/configuration_auto.py:321]: There is no `UPERNET_PRETRAINED_CONFIG_ARCHIVE_MAP`. Let me know if you want to add it in `configuration_upernet.py` and keep this line untouched here, @NielsRogge

Comment (NielsRogge) [reply src/transformers/models/auto/configuration_auto.py:321]: Tbh I've never understood what UPERNET_PRETRAINED_CONFIG_ARCHIVE_MAP is used for 😅

**Review by NielsRogge — APPROVED** [review]
LGTM!

File: src/transformers/models/auto/configuration_auto.py (hunk 1 above)
Comment (ydshieh) [reply src/transformers/models/auto/configuration_auto.py:321]: The only place I find is 
https://github.com/huggingface/transformers/blob/1b37fb5e17add5e65f7076f4e88225e11156c107/src/transformers/commands/add_new_model_like.py#L690
which uses `CONFIG_ARCHIVE_MAP_MAPPING_NAMES` to get model names. Missing some entry won't fail the method though.

@sgugger I guess you will say it's not worth adding `upernet` to `CONFIG_ARCHIVE_MAP_MAPPING_NAMES`, considering the low usage (for now) and the fact of we don't want to maintain such archive maps in the future, rigth?

Comment (sgugger) [reply src/transformers/models/auto/configuration_auto.py:321]: If it's not added then `ALL_PRETRAINED_CONFIG_ARCHIVE_MAP` will be wrong (in the sense it won't have all architectures) but I don't know if we care about that.

**Review by sgugger — APPROVED** [review]
Thanks for the fix!

**Comment by ydshieh** [comment]
I will merge as it is - don't think it matters (at least not at this moment), and eventually this is going to be deprecated.

**Pull request merged**